src/library/scala/collection/Factory.scala

```diff
@@ -312,7 +312,10 @@ object SeqFactory {
     def get: UnapplySeqWrapper[A] = this
     def lengthCompare(len: Int): Int = c.lengthCompare(len)
     def apply(i: Int): A = c(i)
-    def drop(n: Int): scala.Seq[A] = c.view.drop(n).toSeq
+    def drop(n: Int): scala.Seq[A] = c match {
+      case seq: scala.Seq[A] => seq.drop(n)
+      case _                 => c.view.drop(n).toSeq
```

Comment:
In which situation is it actually a good idea to go through a view here ?

Reply:
going through a view avoids building an unnecessary intermediate collection (with the same type as `c`)

Reply:
Right, but that's only true for types which aren't Seq but use UnapplySeqWrapper. Which types are those exactly? Do we have test cases that exercise this code path ?

Reply:
we don't. and they will still be forced.

I think saying that it ought to be `LazyList.from(c.view.drop(n))` (or `LazyList.from(c).drop(n)`) to preserve laziness is valid

Reply:
I can hear @dwijnand's voice in my head asking if this is really all necessary and the best design

Reply:
> Right, but that's only true for types which aren't Seq but use UnapplySeqWrapper. Which types are those exactly? 

All the `mutable.Seq` leaves.